### PR TITLE
Separate modificationtimestamps logic

### LINKED ISSUE #11: Separate ModificationTime logic from OSHxxx classes
Reduce the complexity of the OSH Classes and separate business logic from data container.

### PR DESCRIPTION
**Pull request by rtroilo**
this PR separate the logic of different timestamp calculations from OSHxxx classes to a new utility OSHDBTimestamps class.
this slims down the OSHxxx interfaces and groups logic for timestamp calculation to a single class.
this PR deprecates all methods concerning timestamps in th OSHxxx interface in favor of the new utility class.
this resolves #11

### CODE REVIEW

File: oshdb/src/main/java/org/heigit/bigspatialdata/oshdb/util/OSHDBTimestamps.java [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+import org.heigit.bigspatialdata.oshdb.osm.OSMEntity;
+import org.heigit.bigspatialdata.oshdb.osm.OSMMember;
+import org.heigit.bigspatialdata.oshdb.osm.OSMRelation;
+import org.heigit.bigspatialdata.oshdb.osm.OSMWay;
+
+import com.google.common.collect.Lists;
+
+public abstract class OSHDBTimestamps {
+	
+	public static <OSM extends OSMEntity> SortedMap<OSHDBTimestamp, OSM> getByTimestamps(Iterable<OSM> versions, List<OSHDBTimestamp> byTimestamps) {
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/bigspatialdata/oshdb/util/OSHDBTimestamps.java]: this file is indented using tabs instead of spaces

Comment (tyrasd) [reply oshdb/src/main/java/org/heigit/bigspatialdata/oshdb/util/OSHDBTimestamps.java]: oops, I see this was fixed in the next commit. sorry

**Comment by tyrasd** [comment]
> So just renamed OSHDBTimestamps to OSHEntityTimestamps

Let's do `OSHEntities` (I've already amended the changeset above)?

File: oshdb/src/main/java/org/heigit/bigspatialdata/oshdb/osh/OSHEntities.java [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+  /**
+   * Returns all timestamps at which this entity (or one or more of its child entities) has been
+   * modified and matches a given condition/filter.
+   * 
+   * @param osh the osh entity to work on
+   * @param osmEntityFilter only timestamps for which the entity matches this filter are returned
+   * @return a list of timestamps where this entity has been modified
+   */
+  public static List<OSHDBTimestamp> getModificationTimestamps(OSHEntity<? extends OSMEntity> osh,
```
[the remark this answers is not in the document]
Comment (rtroilo) [reply oshdb/src/main/java/org/heigit/bigspatialdata/oshdb/osh/OSHEntities.java]: If it is only valid for OSHNodes then lets us nail the parameter of this method to OSHNode, and create new getModificationTimestamps with OSHEntity which do a "instance of" check

[the remark this answers is not in the document]
Comment (tyrasd) [reply oshdb/src/main/java/org/heigit/bigspatialdata/oshdb/osh/OSHEntities.java]: :+1: that's done in https://github.com/GIScience/oshdb/pull/12/commits/d9b306c5274a1460bbac6baeef0adf94ef30b770#diff-926cc3e62bbf086082df362afed220d0R178

**Pull request merged**